product-service/src/functions/createProduct/handler.ts

```diff
@@ -0,0 +1,33 @@
+import { APIGatewayProxyEvent } from "aws-lambda";
+import { CreateProductDTO } from "../../types/product";
+
+import * as productService from "../../services/product.service";
+import { lambdaHandler, formatResponse } from "../../utils/handler.utils";
+import { HttpCode, HttpError } from "../../utils/http.utils";
+
+function validateBody(body: CreateProductDTO) {
+  const { title, description, count, price } = body;
+  const errors: string[] = [];
+
+  if (!title || typeof title !== "string") errors.push("Title is missing or invalid.");
```

Comment:
suggestion (on future): it would be helpful to use some libs for auto validate payloads with json schema

Reply:
Yes, it is in my mind for future improvements. I am not sure but probably I will use some validation packages like joi or yup etc.
Thanks for reminding me.